backend/logic/courses/courses.js

```diff
@@ -1,19 +1,153 @@
-const mysql = require('../../sql/connection');
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format");
 
-async function queryCourseByTag(tag) {
-    if (!tag) {
-        throw Error("Undefined tag")
+/**
+ * Returns a list of courses matching the tag
+ * @author Alex Lam
+ * @param {string} tag
+ * @returns A list of courses in JSON format
+ * @throws Undefined tag if tag is null
+ *         error if MySQL connection failed
+ */
+var queryCourseByTag = async function queryCourseByTag(tag) {
+  if (!tag) {
+    throw Error("Undefined tag");
+  }
+
+  let connection = await mysql.getNewConnection();
```

Comment:
Should this be in the try statement?

Reply:
Nope @AlexLam1997 changed the getNewConnection() so that it throws an error with the error message